src/App/Controller/ArticleController.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use App\Entity\Article;
+use App\Entity\ArticleVersion;
+use App\Form\Article\EditArticleType;
+
+class ArticleController extends Controller
+{
+    public function showAction($slug)
+    {
+        $article = $this->getDoctrine()->getRepository('App:Article')->findOneBySlug($slug);
+
+        if (null === $article) {
+            throw $this->createNotFoundException();
+        }
+
+        return $this->render('App:Article:show_article.html.twig', [
+            'article' => $article,
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function addAction(Request $request, $proposalSlug)
+    {
+        $proposal = $this->getDoctrine()->getRepository('App:Proposal')->findOneBySlug($proposalSlug);
+
+        if (null === $proposal) {
+            throw $this->createNotFoundException();
+        }
+
+        $this->denyAccessUnlessGranted('edit', $proposal);
+
+        $article          = new Article();
+        $numberOfArticles = $proposal->getNumberOfArticles();
+        $form             = $this->createForm(EditArticleType::class, $article);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $article
+                ->setProposal($proposal)
+                ->setNumber($numberOfArticles + 1)
+                ->snapshot()
+            ;
+
+            $proposal
+                ->incrementVersionNumber()
```

Comment:
Same?

Reply:
Ok. Everything has been put in the setProposal method.